pkg/search/postgres/common.go

```diff
@@ -325,8 +326,35 @@ func (p *parsedPaginationQuery) AsSQL() string {
 	return paginationSB.String()
 }
 
-func standardizeQueryAndPopulatePath(q *v1.Query, schema *walker.Schema, queryType QueryType) (*query, error) {
+func enrichQueryWithSACFilter(ctx context.Context, q *v1.Query, schema *walker.Schema, queryType QueryType) (*v1.Query, error) {
+	switch queryType {
+	case DELETE:
+		if schema.PermissionChecker != nil {
+			if ok, err := schema.PermissionChecker.WriteAllowed(ctx); err != nil {
+				return nil, err
+			} else if !ok {
+				return nil, sac.ErrResourceAccessDenied
+			}
+			return q, nil
+		}
+		sacFilter, err := GetReadWriteSACQuery(ctx, schema.ScopingResource)
+		if err != nil {
+			return nil, err
+		}
+		pagination := q.GetPagination()
+		query := searchPkg.ConjunctionQuery(sacFilter, q)
+		query.Pagination = pagination
+		return query, nil
+	}
+	return q, nil
+}
```

Comment:
This file has grown significantly already so I recommend moving this to `sac.go` file.

Reply:
I agree with the suggestion to move it to `sac.go`